### DISCUSSION TITLE
AI Playground Interface Mockup

### DISCUSSION THREAD

**Discussion by justinh-rahb** [open-webui/open-webui#928] [Show and tell]
I've noticed a gap in the availability of a versatile, open-source "AI Playground" interface that can integrate with various LLMs (local or APIs)—a staple feature of many AI platforms. To address this, I've put together a simple mockup [using Vercel's v0](https://v0.dev) to conceptualize an **Open WebUI Playground**. My hope is that this sparks discussions and ideas for development within the community. I look forward to any feedback or interest in bringing this concept to fruition.

![Screenshot 2024-02-26 at 10 18 22 AM](https://github.com/open-webui/open-webui/assets/52832301/44d052d7-b9eb-4ca9-afae-b97581fc92a7)

https://v0.dev/t/1cx3UaHjmhs

**Comment by justinh-rahb** [comment]
An optimal approach would be to design an additional interface that operates parallel to the existing frontend. This can be done without making substantial changes to the backend, leveraging the established API communication.

**Comment by tbendien** [comment]
This is excellent @justinh-rahb 
It would be very useful to build a capability into the Playground that allows you to set a system prompt and then do a run of test prompts (from a csv file for example) against different models to test output for a given use case.

You could then run the output against a chat optimized  model that would summarize the difference in the output eg for bias, model role compliance, quality of the output etc.

That is also a capability that AI devlopers could provide as a billable service if they have a tool that can do this easliy.

Gen AI is all about understanding and defining use cases and then testing them against different models with prompt engineering.

So a tool like this would be great for the AI community. I am not a coder/developer, but can share use cases for the Playground and test the use cases :-)

**Reply by justinh-rahb** [reply]
> I am not a coder/developer, but can share use cases for the Playground and test the use cases :-)

Absolutely, please do! Good _user stories_ are important for planning something like this out.

**Comment by justinh-rahb** [comment]
Some more massaging of my prompt in v0 yielded this:

### A:
<img width="1462" alt="Screenshot 2024-02-27 at 2 08 29 AM" src="https://github.com/open-webui/open-webui/assets/52832301/a277102e-5c10-42c7-a320-2a2037e0bd6d">

### B:
<img width="1462" alt="Screenshot 2024-02-27 at 2 01 44 AM" src="https://github.com/open-webui/open-webui/assets/52832301/a8a5e6f2-0a64-4f42-9cb1-6e4a36d56a3b">

https://v0.dev/t/2ffP0nh6FKt

**Comment by justinh-rahb** [comment]
Last concept art I'm gonna do haha

![Screenshot 2024-02-27 at 10 46 22 AM](https://github.com/open-webui/open-webui/assets/52832301/4f630e62-c6e2-4393-8872-054cd7929fe3)

https://v0.dev/r/aqQdwuGJoYt
